src/Penrose/Substance.hs

```diff
@@ -152,12 +156,27 @@ convertDeclList t vars var = vars ++ [Decl t var]
 preludesToDeclarations :: (Var,T) -> SubStmt
 preludesToDeclarations (v,t) = (Decl t v)
 
-
 -- | 'substanceParser' is the top-level parser function. The parser contains a list of functions
 --    that parse small parts of the language. When parsing a source program, these functions are invoked in a top-down manner.
-substanceParser :: VarEnv -> BaseParser [SubStmt]
+substanceParser :: VarEnv -> BaseParser [SubStmt']
 substanceParser env = evalStateT substanceParser' $ Just env
-substanceParser' = between scn eof subProg -- Parse all the statemnts between the spaces to the end of the input file
+substanceParser' = between scn eof subProg' -- Parse all the statemnts between the spaces to the end of the input file
+
+subProg' :: Parser [SubStmt']
+subProg' = subStmt' `sepEndBy` newline'
+
+subStmt' :: Parser SubStmt'
+subStmt' = tryChoice [declBind,
+                      singleSubStmt]
+
+declBind :: Parser SubStmt'
+declBind = do t <- tParser
+              var <- varParser
+              rword ":="
+              DeclBind t var <$> exprParser
+
+singleSubStmt :: Parser SubStmt'
```

Comment:
Surprised that this compiled? The `do` is probably redundant here. If you agree with the AST change, we don't even have to write these functions altogether.